src/providers/export/tensorFlowRecords/tensorFlowBuilder.ts

```diff
@@ -0,0 +1,148 @@
+import { TFRecordsImageMessage, Features, Feature, FeatureList,
+    BytesList, Int64List, FloatList } from "./tensorFlowRecordsProtoBuf_pb";
+import { crc32c, maskCrc, getInt64Buffer, getInt32Buffer, textEncode } from "./tensorFlowHelpers";
+
+/**
+ * @name - TFRecords Builder Class
+ * @description - Create a TFRecords object
+ */
+export class TFRecordsBuilder {
+    /**
+     * @records - An Array of TFRecord Buffer created with releaseTFRecord()
+     * @description - Return a Buffer representation of a TFRecords object
+     */
+    public static releaseTFRecords(records: Buffer[]): Buffer {
+        return Buffer.concat(records.map((record) => {
+            const length = record.length;
+
+            // Get TFRecords CRCs for TFRecords Header and Footer
+            const bufferLength = getInt64Buffer(length);
+            const bufferLengthMaskedCRC = getInt32Buffer(maskCrc(crc32c(bufferLength)));
+            const bufferDataMaskedCRC = getInt32Buffer(maskCrc(crc32c(record)));
+
+            // Concatenate all TFRecords Header, Data and Footer buffer
+            return Buffer.concat([bufferLength,
+                                  bufferLengthMaskedCRC,
+                                  record,
+                                  bufferDataMaskedCRC]);
+        }));
+    }
+
+    private features: Features;
+
+    constructor() {
+        this.features = new Features();
+    }
+
+    /**
```

Comment:
The signatures of these methods are very similar.  What do you think of creating an enum or similar `FeatureType` with the different features and using a single method with signature like 

`addFeature(featureType: FeatureType, key: string, value: string | number | Uint8Array)` ?